tsl/src/bgw_policy/job.c

```diff
@@ -475,8 +475,8 @@ job_execute(BgwJob *job)
 		PushActiveSnapshot(GetTransactionSnapshot());
 	}
 
-	name = list_make2(makeString(NameStr(job->fd.proc_schema)),
-					  makeString(NameStr(job->fd.proc_name)));
+	name = list_make2(makeString(pstrdup(NameStr(job->fd.proc_schema))),
+					  makeString(pstrdup(NameStr(job->fd.proc_name))));
```

Comment:
why are you copying those strings? they are only used for LookupFuncName

Reply:
makeString doesn't take const parameters. We could cast, but probably safer to `pstrdup`

Reply:
This seems to be an omission on the behalf of `makeString` (it is `const` for all practical purposes). IMHO, you can use a cast here instead of copying the string.

Reply:
There is no guarantee, generally, that makeString strings aren't modified, since they are often part of a plan tree. We can check every case to make sure that casting is safe, but I see no reason why we should do that.

Reply:
Have I missed something? AFAICT, `name` is only used in the `LookupFuncName` below and `makeString` does not do anything funny with them before that (like saving them away in some cache).